### PR TITLE
[Plugin] Implement OpenCV-mini

### PR DESCRIPTION
**Pull request by dannypsnl**
### transform
1. `normalize`: not `cv::normalize`, from https://github.com/WasmEdge/WasmEdge/commit/77051da4995d7318d91a82102a72ce2557151764#diff-3333d926ca87cf4285bfcd6deae45ee310307be66fca8a4ca6f0f8a946743fccR50-R54
2. `blur`: blurring function
3. `bilinear_sampling`: bilinear sampler
### read/write
1. `imdecode`: from image from instance buffer
2. `imencode`: put image into instance buffer
4. `imwrite`: write image into file
### misc
1. `imshow`: show image (broken on MacOS)
2. `waitkey`: wait any keypress

----

- Example: https://github.com/second-state/opencvmini-example
- NOTE: since #2403 is failed to trigger CI now, I reopen this PR

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by hydai** [comment]
Please fix the failed CI.

**Comment by dannypsnl** [comment]
failed:
```
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp: In member function ‘bool cv::CvCaptureCAM_V4L::autosetup_capture_mode_v4l2()’:
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:276:29: error: ‘v4l2_fourcc_be’ was not declared in this scope; did you mean ‘v4l2_fourcc’?
  276 | #define V4L2_PIX_FMT_Y16_BE v4l2_fourcc_be('Y', '1', '6', ' ')
      |                             ^~~~~~~~~~~~~~
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:620:13: note: in expansion of macro ‘V4L2_PIX_FMT_Y16_BE’
  620 |             V4L2_PIX_FMT_Y16_BE,
      |             ^~~~~~~~~~~~~~~~~~~
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp: In member function ‘bool cv::CvCaptureCAM_V4L::convertableToRgb() const’:
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:276:29: error: ‘v4l2_fourcc_be’ was not declared in this scope; did you mean ‘v4l2_fourcc’?
  276 | #define V4L2_PIX_FMT_Y16_BE v4l2_fourcc_be('Y', '1', '6', ' ')
      |                             ^~~~~~~~~~~~~~
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:680:10: note: in expansion of macro ‘V4L2_PIX_FMT_Y16_BE’
  680 |     case V4L2_PIX_FMT_Y16_BE:
      |          ^~~~~~~~~~~~~~~~~~~
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp: In member function ‘void cv::CvCaptureCAM_V4L::v4l2_create_frame()’:
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:276:29: error: ‘v4l2_fourcc_be’ was not declared in this scope; did you mean ‘v4l2_fourcc’?
  276 | #define V4L2_PIX_FMT_Y16_BE v4l2_fourcc_be('Y', '1', '6', ' ')
      |                             ^~~~~~~~~~~~~~
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:729:14: note: in expansion of macro ‘V4L2_PIX_FMT_Y16_BE’
  729 |         case V4L2_PIX_FMT_Y16_BE:
      |              ^~~~~~~~~~~~~~~~~~~
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp: In member function ‘void cv::CvCaptureCAM_V4L::convertToRgb(const cv::Buffer&)’:
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:276:29: error: ‘v4l2_fourcc_be’ was not declared in this scope; did you mean ‘v4l2_fourcc’?
  276 | #define V4L2_PIX_FMT_Y16_BE v4l2_fourcc_be('Y', '1', '6', ' ')
      |                             ^~~~~~~~~~~~~~
/__w/WasmEdge/WasmEdge/opencv/modules/videoio/src/cap_v4l.cpp:1753:10: note: in expansion of macro ‘V4L2_PIX_FMT_Y16_BE’
 1753 |     case V4L2_PIX_FMT_Y16_BE:
      |          ^~~~~~~~~~~~~~~~~~~
[1192/1735] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/container_avi.cpp.o
[1193/1735] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_obsensor_capture.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by dannypsnl** [comment]
> Please fix the failed CI.

Fixed, I just pick a stable version of OpenCV

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -394,8 +402,9 @@ jobs:
       - name: Build and install dependencies
         run: |
           yum update -y
-          yum install -y zlib-devel zlib-static
+          yum install -y zlib-devel zlib-static cmake devtoolset-8 wget unzip
```
Comment (hydai) [inline .github/workflows/release.yml:405]: We have devtoolset-11 on x86_64 and devtoolset-10 on aarch64. Do we still need devtoolset-8?

File: plugins/CMakeLists.txt [hunk 2]
```diff
@@ -45,6 +45,15 @@ if(WASMEDGE_PLUGIN_PROCESS)
   endif()
 endif()
 
+if(WASMEDGE_PLUGIN_HTTPSREQ)
```
Comment (hydai) [inline plugins/CMakeLists.txt:48]: There is no HTTPSREQ anymore.

File: plugins/wasmedge_opencvmini/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+target_compile_options(wasmedgePluginWasmEdgeOpenCVMini
+  PUBLIC
+  -DWASMEDGE_PLUGIN
+)
+
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  target_link_options(wasmedgePluginWasmEdgeOpenCVMini
+    PUBLIC
```
Comment (hydai) [inline plugins/wasmedge_opencvmini/CMakeLists.txt:20]: This does nothing.

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  auto *MemInst = Frame.getMemoryByIndex(0);
+  auto OutSpan = MemInst->getSpan<uchar>(BufPtr, BufLen);
+  if (unlikely(OutSpan.size() != BufLen)) {
+    spdlog::error("[WasmEdge-OpenCVMini] "sv
+                  "Failed when accessing the image target buffer memory."sv);
+    return Unexpect(ErrCode::Value::HostFuncError);
+  }
+
+  std::vector<uchar> WriteTo;
+  cv::imencode(".jpg", *Img, WriteTo);
```
Comment (hydai) [inline plugins/wasmedge_opencvmini/opencvmini_func.cpp:110]: Is it possible to expose the image types, so users can decide which format they want?

**Review by hydai — CHANGES_REQUESTED** [review]

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                                             uint32_t Delay) {
+  cv::waitKey(static_cast<int>(Delay));
+  return {};
+}
+
+Expect<uint32_t> WasmEdgeOpenCVMiniBlur::body(const Runtime::CallingFrame &,
+                                              uint32_t SrcMatKey) {
+  cv::Mat Dst;
+  if (auto Src = Env.getMat(SrcMatKey); Src) {
+    cv::blur(*Src, Dst, cv::Size(5, 5));
```
[the remark this answers is not in the document]
Comment (dannypsnl) [reply plugins/wasmedge_opencvmini/opencvmini_func.cpp:63]: Oh, indeed. That's kernel width and kernel height

File: plugins/wasmedge_opencvmini/opencvmini_func.cpp (hunk 4 above)
Comment (dannypsnl) [reply plugins/wasmedge_opencvmini/opencvmini_func.cpp:110]: I think we just reexport this string parameter?

Comment (dannypsnl) [reply plugins/wasmedge_opencvmini/opencvmini_func.cpp]: done

**Review by hydai — APPROVED** [review]
LGTM

**Pull request merged**